src/main/java/edu/unc/lib/boxc/migration/cdm/status/SourceFilesSummaryService.java

```diff
@@ -91,6 +92,96 @@ public int totalFilesInProject() {
+                        .withHeader(SourceFilesInfo.CSV_HEADERS)
+                        .withTrim());
+        ) {
+            for (CSVRecord csvRecord : csvParser) {
+                allEntries.add(csvRecord);
+            }
+        } catch (IOException e) {
+            log.error("Failed to list files", e);
+        }
+        return allEntries;
+    }
+
+    /**
+     * @return load list of CSV records with source file populated
+     */
+    public List<CSVRecord> loadPopulatedEntries(Path mappingPath) {
+        List<CSVRecord> populatedEntries = loadAllEntries(mappingPath);
+
+        return populatedEntries.stream().filter(entry -> !entry.get(SOURCE_FILE_FIELD).isEmpty())
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * @return list of new CSV records with source file populated
+     */
+    public List<CSVRecord> getListNewEntries() {
+        List<CSVRecord> previousMappings = new ArrayList<>();
+        if (previousStatePopulatedFiles != null) {
+            previousMappings = previousStatePopulatedFiles;
+        }
+        List<CSVRecord> newMappings = loadPopulatedEntries(getNewMappingPath());
+
+        if (!previousMappings.isEmpty()) {
+            Set<String> previousIdentifiers = getIdentifierSet(previousMappings);
+            Set<String> newIdentifiers = getIdentifierSet(newMappings);
+            newIdentifiers.removeAll(previousIdentifiers);
+            return newMappings.stream().filter(entry -> newIdentifiers.contains(entry.get(ID_FIELD)))
+                    .collect(Collectors.toList());
+        } else {
+            return newMappings;
+        }
+    }
+
+    private Set<String> getIdentifierSet(List<CSVRecord> mappings) {
+        return mappings.stream().map(entry -> entry.get(ID_FIELD)).collect(Collectors.toSet());
+    }
+
+    /**
+     * @return sample list of new files mapped (every nth file)
+     */
+    public List<CSVRecord> sampleListNewFiles() {
+        List<CSVRecord> completeListNewFiles = getListNewEntries();
+        List<CSVRecord> sampleListNewFiles;
+
+        // select every nth entry
+        int interval = completeListNewFiles.size() / sampleSize;
+
+        sampleListNewFiles = IntStream.range(0, completeListNewFiles.size())
+                .filter(n -> n % interval == 0)
+                .mapToObj(completeListNewFiles::get)
+                .collect(Collectors.toList());
+
+        return sampleListNewFiles;
+    }
+
+    /**
+     * @return sample list of files mapped (if number of mapped files > sample size) or list of all files mapped
+     */
+    public List<CSVRecord> listNewFiles() {
+        List<CSVRecord> listNewFiles = new ArrayList<>();
+        if (Files.exists(getNewMappingPath())) {
+            if (loadPopulatedEntries(getNewMappingPath()).size() > sampleSize) {
+                listNewFiles = sampleListNewFiles();
+            } else {
+                listNewFiles = loadPopulatedEntries(getNewMappingPath());
```

Comment:
After looking at some of the tests, i'm realizing this should probably be calling `getListNewEntries`, otherwise the sample will include previously mapped entries when the total number of mapped entries in the new file is less than the sample size. I think we consistently want the listed files to be just the newly added stuff.